File: src/main/java/com/epam/mentoring/kotlin/util/DogBreedDatabasePopulator.java
package com.epam.mentoring.kotlin.util;

import com.epam.mentoring.kotlin.service.DogBreedService;
import com.epam.mentoring.kotlin.rest.DogBreedApiClient;
import jakarta.annotation.PostConstruct;
import java.util.List;
import java.util.Map;
import org.springframework.stereotype.Component;

@Component
public class DogBreedDatabasePopulator {
    private final DogBreedApiClient dogBreedApiClient;
    private final DogBreedService dogBreedService;

    public DogBreedDatabasePopulator(final DogBreedApiClient dogBreedApiClient, final DogBreedService dogBreedService) {
        this.dogBreedApiClient = dogBreedApiClient;
        this.dogBreedService = dogBreedService;
    }

    @PostConstruct
    public void initializeDatabase() throws Exception {
        if (dogBreedService.getBreeds().isEmpty()) {  //findAll();
            Map<String, List<String>> breeds = dogBreedApiClient.getBreeds(); //call https://dog.ceo/api/breeds/list/all";

            dogBreedService.save(breeds);
        }
    }

}

File: src/main/java/com/epam/mentoring/kotlin/util/DogBreedDatabasePopulator.kt
package com.epam.mentoring.kotlin.util

import com.epam.mentoring.kotlin.rest.DogBreedApiClient
import com.epam.mentoring.kotlin.service.DogBreedService
import kotlinx.coroutines.runBlocking
import org.springframework.context.annotation.Profile
import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.context.event.EventListener

import org.springframework.stereotype.Component

@Profile("dev")
@Component
open class DogBreedDatabasePopulator(private val dogBreedApiClient: DogBreedApiClient, val dogBreedService: DogBreedService) {

// to run initializeDatabase() only after all Beans were created
    @EventListener(ContextRefreshedEvent::class)
    fun onApplicationEvent(event: ContextRefreshedEvent) {
        runBlocking {
            initializeDatabase()
        }
    }

    @Throws(Exception::class)
    suspend fun initializeDatabase(): Unit {
        if (dogBreedService.getAllBreeds().isEmpty()) {
            val breeds: Map<String, List<String>> = dogBreedApiClient.getBreeds();

            dogBreedService.saveAll(breeds);
         }
    }
}